src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -237,11 +332,19 @@ static void GetAddrInfo(ResolveContext * sdCtx)
 
     for (auto & interface : sdCtx->interfaces)
     {
-        auto interfaceId = interface.first;
+        auto interfaceId = interface.first.interfaceId;
         auto hostname    = interface.second.fullyQualifiedDomainName.c_str();
         auto sdRefCopy   = sdCtx->serviceRef; // Mandatory copy because of kDNSServiceFlagsShareConnection
-        auto err = DNSServiceGetAddrInfo(&sdRefCopy, kGetAddrInfoFlags, interfaceId, protocol, hostname, OnGetAddrInfo, sdCtx);
-        VerifyOrReturn(kDNSServiceErr_NoError == err, sdCtx->Finalize(err));
+
+        if (!interface.second.isDNSLookUpRequested)
```

Comment:
Why is this check not the very first thing in the loop, with a `continue` if `isDNSLookUpRequested`?

Reply:
done.